docs/advanced/dataprovider.md

```diff
@@ -157,18 +108,214 @@ where `"Data1"`, `"Data2"`, `"Data3"` in this example is the tests unique identi
 2. In the test, that use our provider, for example
 
 ```
-@Test(dataProvider = "dataProvider")
+@Test(dataProvider = "DP1")
 public void testDataProvider(...) {
     ...
 }
 ```
 
-when you receive data from the data provider, you must name the method parameter 
+when you receive data from the data provider, you must name the method parameter
 that receives a TUID as `TUID`, for example:
 
 ```
-@Test(dataProvider = "dataProvider")
+@Test(dataProvider = "DP1")
 public void testDataProvider(String TUID, String data, ...) {
     ...
 }
 ```
+
+
+## Custom data providers
+
+### Pre requirements
+
+To use custom provider feature, add certain dependency to your pom.xml
+
+```xml
+<dependency>
+    <groupId>com.zebrunner</groupId>
+    <artifactId>carina-dataprovider</artifactId>
+    <version>RELEASE</version>
+</dependency>
+```
+
+Visit our [github](https://github.com/zebrunner/carina-dataprovider/releases) to check the latest version of carina-dataprovider.
+
+### Implementation
+Carina test framework provides a possibility to write all tests with data providers in one place, including parametrization using external XLS/CSV spreadsheets. First of all, you need to declare a test class that implements `IAbstractTest.java` and `IAbstractDataProvider.java`.
```

Comment:
If we set class we should set full path to this class, for example com.zebrunner.carina.dataprovider.IAbstractDataProvider